Demo/Sources/Views/PlaybackSlider.swift

```diff
@@ -0,0 +1,118 @@
+            progressBar(valueWidth: valueWidth, bufferWidth: bufferWidth)
+            text(maximumValueText)
+        }
+        .frame(maxWidth: .infinity, minHeight: 10, maxHeight: 10)
+    }
+
+    init(progressTracker: ProgressTracker, minimumValueText: String? = nil, maximumValueText: String? = nil) {
+        self.progressTracker = progressTracker
+        self.minimumValueText = minimumValueText
+        self.maximumValueText = maximumValueText
+        _value = Binding(progressTracker, at: \.progress)
+        _buffer = Binding(progressTracker, at: \.loaded)
+    }
+
+    @ViewBuilder
+    private func text(_ text: String?) -> some View {
+        if let text {
+            Text(text)
+                .font(.caption)
+                .monospacedDigit()
+        }
+    }
+
+    @ViewBuilder
+    private func rectangle(opacity: Double = 1, width: CGFloat? = nil) -> some View {
+        Rectangle()
+            .foregroundColor(.white)
+            .opacity(opacity)
+            .frame(width: width)
+    }
+
+    @ViewBuilder
+    private func progressBar(valueWidth: CGFloat, bufferWidth: CGFloat) -> some View {
+        GeometryReader { geometry in
+            ZStack(alignment: .leading) {
+                rectangle(opacity: 0.1)
+                rectangle(opacity: 0.3, width: bufferWidth)
+                rectangle(width: valueWidth)
+            }
+            .gesture(dragGesture(geometry: geometry))
+            .onChange(of: value) { onValueChanged($0, geometry: geometry) }
+            .onChange(of: buffer) { onBufferChanged($0, geometry: geometry) }
+        }
+        .cornerRadius(5)
+        .scaleEffect(y: isDragging ? 1.5 : 1.0)
+        .animation(.smooth(duration: 0.4), value: isDragging)
+    }
+
+    private func dragGesture(geometry: GeometryProxy) -> some Gesture {
+        DragGesture()
+            .onChanged { gesture in
+                isDragging = true
+                let width = geometry.size.width
+                let translation = gesture.translation.width
+                valueWidth = (previousValueWidth + translation).clamped(to: 0...width)
+                value = Float(valueWidth / width)
+            }
+            .onEnded { _ in
+                isDragging = false
+                previousValueWidth = valueWidth
+            }
+    }
+
+    func onValueChanged(_ value: Float, geometry: GeometryProxy) {
+        let width = geometry.size.width
+        valueWidth = CGFloat(value) * width
+        if !isDragging {
+            previousValueWidth = valueWidth
+        }
+    }
+
+    private func onBufferChanged(_ buffer: Float, geometry: GeometryProxy) {
+        let width = geometry.size.width
+        let widthToApply = CGFloat(buffer) * width
+        if widthToApply.isNormal {
+            bufferWidth = widthToApply
+        }
+    }
+}
+
+struct PlaybackSlider_Previews: PreviewProvider {
+    static var previews: some View {
+        PlaybackSlider(progressTracker: .init(interval: .zero))
```

Comment:
The preview background should be forced to black (or the preview itself to dark mode) so that the slider is visible.